aggie-coding-club/Rev-Registration
# Save all schedules + selected schedule

## Description

This saves _all_ schedules, adds the saving of the selected schedule, and renames "save schedule" to "lock schedule" throughout the frontend.

Note that we weren't previously linting `user_sessions`, so 8253482  also adds that + fixes all of the failing linting rules.

This will probably be easiest to review if you ignore the changes from 8253482 and 29cbefc (which the only valuable change from it is that I updated one of the tests to also check for the selected schedule)

## How to test

Add a bunch of schedules, lock some and leave other unlocked, select one, then refresh & ensure everything is restored correctly.

Also confirm that coming from the old session that the new one still works.

## Screenshots

![save-schedules](https://user-images.githubusercontent.com/7295783/133003144-01086dfc-52ad-4ce2-89e6-eba034829c58.gif)

## Related tasks

Closes #577
